packages/gatsby-recipes/src/providers/gatsby/plugin.js

```diff
@@ -376,9 +454,29 @@ class BabelPluginGetPluginsFromGatsbyConfig {
               return
             }
 
+            // Only allow module.exports = {}
+            // Not things like module.exports = () => ({})
+            if (!t.isObjectExpression(right)) {
```

Comment:
No warnings for bailing on an unsupported structure?

Reply:
I really could remove this because we already check for this anyways in the bootstrap of the site. Added this before checking it.